lib/customisation.nix

```diff
@@ -365,6 +378,35 @@ rec {
     else
       mapAttrs mkAttrOverridable pkgs;
 
+  makeAlias =
+    let
+      currentRelease = lib.versions.majorMinor lib.version;
+    in
+    self: name:
+    {
+      type,
+      package ? null,
+      reason ? null,
```

Comment:
It would be nice to support multiple reasons, if we can ensure the interface remains convenient. I also think that we should ensure the reasons are structured by default rather than free‐form strings, because our existing throws are wildly inconsistent, and it’s much more convenient to write `unmaintained` than `unmaintained in Nixpkgs` but can give the wrong idea about upstream status.

Vague draft: `reason.broken = true; reason.unmaintained = true;` → “it was broken and unmaintained in Nixpkgs”? And then we could also have `reason.insecure = true;` or `reason.insecure = ["CVE1" "CVE2"];` or such, and of course `reason.custom = "I just really don’t like it";`.